src/kaleidoscope/plugin/Qukeys.cpp

```diff
@@ -50,6 +50,14 @@ EventHandlerResult Qukeys::onKeyswitchEvent(Key& key, KeyAddr k, uint8_t key_sta
 
   // Deal with keyswitch state changes.
   if (keyToggledOn(key_state) || keyToggledOff(key_state)) {
+    // If the user rolled over from a non-modifier key to a qukey, let the
+    // release event for that key skip the queue. This prevents unintended
+    // repeat characters for the tapped key, which would otherwise have its
+    // release event delayed.
+    if (keyToggledOff(key_state) && event_queue_.length() == 1 &&
+        k != event_queue_.addr(0) && isModifierKey(key)) {
```

Comment:
Past me was sloppy. The last condition here is wrong, and should be `! isModifierKey(key)`…